<launch>

  <node pkg="tf" type="static_transform_publisher" name="world" args="0 0 0 0 0 0 map world 100" /> 

	<node name="orb_slam2_mono" pkg="orb_slam2_ros" type="orb_slam2_ros_mono" output="screen">
      <remap from="/camera/image_raw" to="/tello/camera/image_raw" />
      <param name="publish_pointcloud" type="bool" value="true" />
      <param name="publish_pose" type="bool" value="true" />
      <param name="localize_only" type="bool" value="false" />
      <param name="reset_map" type="bool" value="false" />
      <!-- static parameters -->
      <param name="use_viewer" type="bool" value="true" />
      <param name="load_map" type="bool" value="false" />
      <param name="map_file" type="string" value="labaratory_rot.bin" />
      <param name="settings_file" type="string" value="$(find orb_slam2_ros)/orb_slam2/config/Tello.yaml" />
      <param name="voc_file" type="string" value="$(find orb_slam2_ros)/orb_slam2/Vocabulary/ORBvoc.txt" />

      <param name="pointcloud_frame_id" type="string" value="map" />
      <param name="camera_frame_id" type="string" value="camera_link" />
      <param name="min_num_kf_in_map" type="int" value="15" />
  </node>

  <!-- <node name="image_muxer" pkg="flock_driver" type="image_muxer.py" output="screen" /> -->

  <node name="cloud_map_saver" pkg="flock_driver" type="cloud_map_saver.py" output="screen">
  <param name="OUT_FILE_PATH" type="string" value="~/Documents/Maps" />
  <param name="CLOUD_TOPIC_NAME" type="string" value="/orb_slam2_mono/map_points" />
  <param name="TRIGGER_TOPIC_NAME" type="string" value="/save_cloud_trigger" />
  <param name="POSE_TOPIC_NAME" type="string" value="/orb_slam2_mono/pose" />
  <param name="CAMERA_TOPIC_NAME" type="string" value="/orb_slam2_mono/debug_image" />


</node>
 <node name="flock_driver_node" pkg="flock_driver" type="flock_driver.py" output="screen" >
    <param name="network_interface" type="string" value="" />
    <param name="ID" type="string" value="" />
 </node>
  <arg name="joy_dev" default="/dev/input/js0" />
  <node pkg="joy" type="joy_node" name="joy_node">
    <param name="dev" value="$(arg joy_dev)" />
    <param name="deadzone" value="0.2" />
  </node>

  <node name="tello_slam_control" pkg="flock_driver" type="tello_slam_control.py" output="screen">
    <param name="~POSE_TOPIC_NAME" type="string" value="/orb_slam2_mono/pose" />
    <param name="ID" type="string" value="" />
  </node>

  <node pkg="flock_driver" type="tello_joy.py" name="joy_teleop" />
  <node pkg="flock_driver" type="auto_node.py" name="calibrator" />
  <node name="override_node" pkg="flock_driver" type="override_node.py" output="screen" launch-prefix="gnome-terminal --command"/>
  <node name="logger" pkg="flock_driver" type="data_logger.py" output="screen"/>
</launch>